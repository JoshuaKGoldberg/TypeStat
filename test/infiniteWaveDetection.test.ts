import { expect, test } from "vitest";

import { checkTestResult } from "../src/tests/testSetup.js";

const cwd = import.meta.dirname;

test("Infinite wave detection", async () => {
	expect.assertions(3);
	await checkTestResult(cwd, "infinite wave detection");
}, 50000);
