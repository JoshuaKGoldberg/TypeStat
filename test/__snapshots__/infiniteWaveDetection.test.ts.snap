// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Infinite wave detection > options 1`] = `
"[
  {
    "cleanups": {
      "suppressTypeErrors": true
    },
    "compilerOptions": {
      "include": [
        "actual.ts"
      ],
      "compileOnSave": false,
      "noImplicitAny": false,
      "noImplicitThis": false,
      "strictNullChecks": false
    },
    "files": {
      "above": "",
      "below": "",
      "renameExtensions": false
    },
    "filters": [],
    "fixes": {
      "importExtensions": false,
      "incompleteTypes": false,
      "missingProperties": false,
      "noImplicitAny": false,
      "noImplicitThis": false,
      "noInferableTypes": false,
      "strictNonNullAssertions": false
    },
    "hints": {
      "react": {
        "propTypes": "whenRequired",
        "propTypesOptionality": "asWritten"
      }
    },
    "include": [
      "actual.ts"
    ],
    "mutators": [
      [
        "./infiniteMutator.cjs",
        null
      ]
    ],
    "output": {},
    "package": {
      "directory": "<rootDir>",
      "file": "<rootDir>/package.json"
    },
    "postProcess": {
      "shell": []
    },
    "projectPath": "<rootDir>/tsconfig.json",
    "types": {}
  }
]"
`;
